### PR TITLE
give friendly error message for Pad + indices case

### LINKED ISSUE #167: Can't make imfilter! for specific indices work.
Hi,
I have a question about how to use `imfilter!` so that is computes only on provided inds. I am having trouble making a simple example work. Here is my code:

```Julia
using ImageFiltering, Images, ComputationalResources

# Create a sample image:
tile_size = 50
board_size = 4 # must be even
mini_board = [zeros(tile_size, tile_size) ones(tile_size, tile_size);
              ones(tile_size, tile_size) zeros(tile_size, tile_size)]
chessboard = repeat(mini_board, outer=(convert(Integer,(board_size/2)), convert(Integer,(board_size/2))))
img = chessboard

# Create a filter:
sigma = 3
filter_size = 21
gaus = ImageFiltering.KernelFactors.gaussian(sigma, filter_size)
kernel = kernelfactors((gaus, gaus))

# params of imfilter function
imgfilt = similar(img)
inds = CartesianIndex(50, 50) # simple inds

# The filtering that errors:
imfilter!(CPU1(Algorithm.FIRTiled()), imgfilt, img, kernel, "symmetric", inds)
```
From the [documentation](https://juliaimages.org/stable/function_reference/#ImageFiltering.imfilter!) I thought this would work, but it doesn't. I get an error like this:
```Julia
ERROR: MethodError: no method matching factorkernel(::Tuple{ImageFiltering.KernelFactors.ReshapedOneD{Float64,2,0,OffsetArrays.OffsetArray{Float64,1,Array{Float64,1
}}},ImageFiltering.KernelFactors.ReshapedOneD{Float64,2,1,OffsetArrays.OffsetArray{Float64,1,Array{Float64,1}}}})
Closest candidates are:
  factorkernel(::AbstractArray{T,2}) where T at /Users/MrTrololord/.julia/packages/ImageFiltering/jhBno/src/imfilter.jl:1493
  factorkernel(::AbstractArray) at /Users/MrTrololord/.julia/packages/ImageFiltering/jhBno/src/imfilter.jl:1489
  factorkernel(::ImageFiltering.Kernel.Laplacian) at /Users/MrTrololord/.julia/packages/ImageFiltering/jhBno/src/imfilter.jl:1490
Stacktrace:
 [1] imfilter!(::CPU1{ImageFiltering.Algorithm.FIRTiled{0}}, ::Array{Float64,2}, ::Array{Float64,2}, ::Tuple{ImageFiltering.KernelFactors.ReshapedOneD{Float64,2,0,O
ffsetArrays.OffsetArray{Float64,1,Array{Float64,1}}},ImageFiltering.KernelFactors.ReshapedOneD{Float64,2,1,OffsetArrays.OffsetArray{Float64,1,Array{Float64,1}}}}, :
:String, ::CartesianIndex{2}) at /Users/MrTrololord/.julia/packages/ImageFiltering/jhBno/src/imfilter.jl:585
 [2] top-level scope at /Users/MrTrololord/Google_Drive/cvut/bakalarka/LAP_julia/test/try4.jl:182
```
--------
Is there perhaps a dispatch problem here?

This is the fuction at `imfilter.jl:585`:
```Julia
function imfilter!(r::AbstractResource, out::AbstractArray, img::AbstractArray, kernel, args...)
    imfilter!(r, out, img, factorkernel(kernel), args...)
end
```
The kernel I use has:
```julia
julia> typeof(kernel) <: ImageFiltering.ProcessedKernel
true
```
Isn't there supposed to be a function that takes this kind of kernel and also has `args..`?

(Disclaimer; I am new to Julia so this might just be me not using the function correctly. Sorry for that.)

### PR DESCRIPTION
**Pull request by johnnychen94**
closes #167

### CODE REVIEW

**Comment by johnnychen94** [comment]
The error seems unrelated. Will check it tomorrow.

**Comment by timholy** [comment]
Hopefully fixed by #174

File: src/imfilter.jl [hunk 1]
```diff
@@ -739,6 +739,10 @@ end
 # cascaded kernels, and even implement multithreadable tiling for FIR
 # filtering.
 
+function imfilter!(::AbstractResource, ::AbstractArray, ::AbstractArray, ::ProcessedKernel, border::AbstractBorder, ::Indices)
+    error("Invalid border strategy `$border`: only `NoPad()` is supported.")
```
Comment (timholy) [inline src/imfilter.jl:743]:
```suggestion
    error("Invalid border strategy `$border`: only `NoPad()` is supported.\n(You have called a stage of `imfilter`'s dispatch hierarchy after border-handling.)")
```

**Comment by timholy** [comment]
Thanks @johnnychen94!

**Pull request merged**